### PR TITLE
chore: creating initial upgrade file

### PR DESCRIPTION
**Pull request by mark1000**
The line
"Normal FW rules > Hierarchical FW rules (functionality is the same, construct is different)"
is ambiguous. What are we trying to say?

### CODE REVIEW

**Comment by daniel-cit** [comment]
Hi @mark1000 , Thanks for the PR.

The lint issues are:

```
Step #1 - "lint": Checking for trailing whitespace
Step #1 - "lint": ./docs/UPGRADE.md:7:Note: There is no in-place upgrade path from v1 to v2. 
Step #1 - "lint": ./docs/UPGRADE.md:15:   hub-and-spoke network are located in 1-org and 3-networks. 
Step #1 - "lint": ./docs/UPGRADE.md:20:   The Terraform can be found in 4-projects, within the infra_pipeline module.  
Step #1 - "lint": ./docs/UPGRADE.md:30:   manually merge changes from V2 into your modified version of V1. 
Step #1 - "lint": Error: Trailing whitespace found in the lines above.
```

**Comment by daniel-cit** [comment]
> Normal FW rules > Hierarchical FW rules (functionality is the same, construct is different)"

Hi @mark1000,  I guess it means something like this:

[Normal Firewall rules](https://cloud.google.com/vpc/docs/firewalls) has been superseded with [Hierarchical FW rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.

**Comment by bharathkkb** [comment]
@mark1000 I replaced normal firewall rules with regular as docs seem to prefer that
https://cloud.google.com/vpc/docs/using-firewall-policies#:~:text=regular%20vpc%20firewall%20rules

**Comment by mark1000** [comment]
Not sure how to reply directly to a comment.
Two things:
- the two types of firewall rules are "VPC firewall rules" and "Hierarchical firewall policy rules".
- HFW rules don't exactly replace VPC firewall rules. The two work together at the moment. Depending on what you want to do, you need both.

File: docs/upgrading_to_v2.0.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+   0.12.x.
+-  V2 introduces a new alternative hub-and-spoke network architecture,
+   described in Section 7.2 of the blueprint. The Terraform to build the
+   hub-and-spoke network are located in 1-org and 3-networks.
+-  In V2, the infrastructure pipeline has transitioned from using Google
+   Container Registry (GCR) to using Google Artifact Registry. Artifact
+   Registry extends the capabilities of GCR as outlined
+   [here](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr#compare).
+   The Terraform can be found in 4-projects, within the infra_pipeline module.
+-  [Some Regular Firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical FW rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.
```
Comment (mark1000) [inline docs/upgrading_to_v2.0.md:21]: [Some VPC firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical firewall policy rules]

Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:21]:
```suggestion
-  Some [VPC firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical firewall policy rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.
```

as per @mark1000 suggestion

File: docs/upgrading_to_v2.0.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+[Changelog](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/CHANGELOG.md).
+
+Note: There is no in-place upgrade path from v1 to v2.
+
+## Breaking Changes
+
+-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
+   0.12.x.
+-  V2 introduces a new alternative hub-and-spoke network architecture,
+   described in Section 7.2 of the blueprint. The Terraform to build the
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:14]:
```suggestion
   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf). The Terraform to build the
```

File: docs/upgrading_to_v2.0.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Upgrade Guidance
+
+Before moving forward with adopting components of V2, please review the list of
+breaking changes below. You can find a list of all changes in the
+[Changelog](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/CHANGELOG.md).
+
+Note: There is no in-place upgrade path from v1 to v2.
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:7]:
```suggestion
**Note:** There is no in-place upgrade path from v1 to v2.
```

File: docs/upgrading_to_v2.0.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
+   0.12.x.
+-  V2 introduces a new alternative hub-and-spoke network architecture,
+   described in Section 7.2 of the blueprint. The Terraform to build the
+   hub-and-spoke network are located in 1-org and 3-networks.
+-  In V2, the infrastructure pipeline has transitioned from using Google
+   Container Registry (GCR) to using Google Artifact Registry. Artifact
+   Registry extends the capabilities of GCR as outlined
+   [here](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr#compare).
+   The Terraform can be found in 4-projects, within the infra_pipeline module.
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:20]: what does `The Terraform` referes to ?

File: docs/upgrading_to_v2.0.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+## Breaking Changes
+
+-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
+   0.12.x.
+-  V2 introduces a new alternative hub-and-spoke network architecture,
+   described in Section 7.2 of the blueprint. The Terraform to build the
+   hub-and-spoke network are located in 1-org and 3-networks.
+-  In V2, the infrastructure pipeline has transitioned from using Google
+   Container Registry (GCR) to using Google Artifact Registry. Artifact
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:17]:
```suggestion
-  In V2, the infrastructure pipeline has transitioned from using 
[Google Container Registry](https://cloud.google.com/container-registry/docs) (GCR) to using [Google Artifact Registry](https://cloud.google.com/artifact-registry/docs). Artifact
```

File: docs/upgrading_to_v2.0.md (hunk 2 above)
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:14]: what does `The Terraform` referes to ? 
are they scripts or code?

File: docs/upgrading_to_v2.0.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+   Registry extends the capabilities of GCR as outlined
+   [here](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr#compare).
+   The Terraform can be found in 4-projects, within the infra_pipeline module.
+-  [Some Regular Firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical FW rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.
+
+## Steps to upgrade codebase
+
+Note: When you run `terraform apply`, expect resources to be deleted and recreated. Be sure
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:25]:
```suggestion
**Note:** When you run `terraform apply`, expect resources to be deleted and recreated. Be sure
```

**Review by daniel-cit — COMMENTED** [review]
I added some suggestions.

The file `5-app-infra/README.md` must be removed from this PR

File: docs/upgrading_to_v2.0.md (hunk 4 above)
Comment (bharathkkb) [reply docs/upgrading_to_v2.0.md:20]: I assume it means the configs are located in those stages, but I think we can remove this.

File: docs/upgrading_to_v2.0.md (hunk 2 above)
Comment (bharathkkb) [reply docs/upgrading_to_v2.0.md]: removed

**Comment by daniel-cit** [comment]
lint failed with trailing whitespace in the GitHub suggestions:

```
Step #1 - "lint": Checking for trailing whitespace
Step #1 - "lint": ./docs/upgrading_to_v2.0.md:14:   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf). 
Step #1 - "lint": ./docs/upgrading_to_v2.0.md:15:-  In V2, the infrastructure pipeline has transitioned from using 
Step #1 - "lint": Error: Trailing whitespace found in the lines above.
```

File: docs/upgrading_to_v2.0.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+**Note:** There is no in-place upgrade path from v1 to v2.
+
+## Breaking Changes
+
+-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
+   0.12.x.
+-  V2 introduces a new alternative hub-and-spoke network architecture,
+   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf). 
+-  In V2, the infrastructure pipeline has transitioned from using
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:15]:
```suggestion
-  In V2, the infrastructure pipeline has transitioned from using
```

File: docs/upgrading_to_v2.0.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+[Changelog](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/CHANGELOG.md).
+
+**Note:** There is no in-place upgrade path from v1 to v2.
+
+## Breaking Changes
+
+-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
+   0.12.x.
+-  V2 introduces a new alternative hub-and-spoke network architecture,
+   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf).
```
Comment (daniel-cit) [inline docs/upgrading_to_v2.0.md:14]:
```suggestion
   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf).
```

**Comment by daniel-cit** [comment]
Hi @bharathkkb  the integration builds are failing because of 
```
Step #0 - "prepare": Error: Error applying IAM policy for organization "REDACTED_ORG_ID": Error setting IAM policy for organization "REDACTED_ORG_ID": googleapi: Error 400: The number of members in the policy (1,501) is larger than the maximum allowed size 1,500., badRequest
Step #0 - "prepare": 
Step #0 - "prepare":   on iam.tf line 37, in resource "google_organization_iam_member" "org_admins_group":
Step #0 - "prepare":   37: resource "google_organization_iam_member" "org_admins_group" {
```

I guess these are deleted service accounts filling the organization policy

**Comment by bharathkkb** [comment]
@daniel-cit just triggered cleanup job, I'll bump the schedule to run twice a week.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**